lang/cs/Org.Apache.REEF.Tests/Functional/Failure/EvaluatorWithRunningTaskImmediatePoisonTest.cs

```diff
@@ -0,0 +1,138 @@
+                .Set(DriverConfiguration.OnContextFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Build();
+        }
+
+        private sealed class PoisonedEvaluatorDriver :
+            BasePoisonedEvaluatorDriver,
+            IObserver<IFailedEvaluator>,
+            IObserver<IActiveContext>
+        {
+            [Inject]
+            private PoisonedEvaluatorDriver(IEvaluatorRequestor requestor) : base(requestor)
+            {
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                var taskConfig = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId)
+                    .Set(TaskConfiguration.Task, GenericType<SleepTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<SleepTask>.Class)
+                    .Set(TaskConfiguration.OnTaskStart, GenericType<PoisonedEventHandler<ITaskStart>>.Class)
+                    .Build();
+
+                var poisonConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                    .BindIntNamedParam<CrashTimeout>("0")
+                    .BindIntNamedParam<CrashMinDelay>("0")
+                    .BindNamedParameter<CrashProbability, double>(GenericType<CrashProbability>.Class, "1.0")
+                    .Build();
+
+                value.SubmitTask(Configurations.Merge(taskConfig, poisonConfig));
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
+                Logger.Log(Level.Info, FailedEvaluatorMessage);
+                if (value.FailedTask.Value == null || !value.FailedTask.IsPresent())
+                {
+                    throw new Exception("No failed Task associated with failed Evaluator");
+                }
+
+                if (value.FailedTask.Value.Id != TaskId)
+                {
+                    throw new Exception("Failed Task ID returned " + value.FailedTask.Value.Id
+                        + ", was expecting Task ID " + TaskId);
+                }
+
+                Logger.Log(Level.Info, "Received all expected failed Tasks.");
+
+                var expectedStr = "expected a single Context with Context ID " + ContextId + ".";
+
+                if (value.FailedContexts == null)
+                {
+                    throw new Exception("No Context was present but " + expectedStr);
+                }
+
+                if (value.FailedContexts.Count != 1)
+                {
+                    throw new Exception("Collection of failed Contexts contains " + value.FailedContexts.Count + " failed Contexts but only " + expectedStr);
+                }
+
+                if (!value.FailedContexts.Select(ctx => ctx.Id).Contains(ContextId))
```

Comment:
FailedContexts.count is 1 in this case, you don't need to use select and contains to verify. FailedContexts[0] should be Ok.